libc/include/signal.yaml

```diff
@@ -69,10 +69,15 @@ functions:
   - name: signal
     standards:
       - stdc
-    return_type: sighandler_t
+    # May the Geneva Convention have mercy on my soul... Why this insanity?
+    # Well: signal returns a function pointer to a function with no return
+    # value and which accepts an int. The parameter list appears on the far
+    # right of the declaration. i.e.
+    # void (*signal(int, void (*)(int)))(int);
+    return_type: void (*
     arguments:
       - type: int
-      - type: sighandler_t
+      - type: void (*)(int)))(int
```

Comment:
is the unmatched `(` intentional?

Reply:
Yes, see the comment. hdrgen just pastes `return_type` + `name` + `(` + arguments + `)`.

Reply:
i still don't understand ... shouldn't that still have matching parens?

Reply:
The result in the generated header (visible at build time, not something I can link to in sources) is:

```c
void (* signal(int, void (*)(int)))(int)
```
which looks correct to me, typechecks (via `clang build/libc/include/signal.h -isystem build/libc/include -fsyntax-only -Wsystem-headers`), and cdecl seems to be able to decode this declaration as I would have expected. https://cdecl.org/?q=void+%28*+signal%28int%2C+void+%28*%29%28int%29%29%29%28int%29